docs/writing/posts/stochastic-software.md

```diff
@@ -0,0 +1,114 @@
+
+| Query                | Count | Quality |
+| -------------------- | ----- | ------- |
+| Personal Data        | 420   | 70%     |
+| Scheduling Questions | 90    | 83%     |
+| Internet Searches    | 20    | 9%      |
+
+We have observed that the quality of internet searches is exceptionally low, while the majority of our queries pertain to personal data. The quality metric is determined by a thumbs-up or thumbs-down rating system. Additionally, scheduling questions exhibit high quality, but their volume is relatively low. Based on this data, we can recommend allocating more time to improve the quality of personal data. Furthermore, due to the low volume of internet searches, we may consider informing users that this service will not be available until we have a better plan.
+
+Here are some examples of recommendations that we can make based on this data:
+
+1. Our performance in internet searches is clearly underwhelming, but the count is quite low.
+2. In the meantime, we can disable this feature, knowing that it won't significantly impact our users.
+3. Personal data queries have a very high volume, but the quality is lacking.
+4. We should focus on building experiments that improve the quality of personal data queries.
+5. Since we can't run a backtest on users thumbs up and thumbs down ratings, we should consider a different metric like embedding reranking scores
+6. If we can show that changing our retrieval system can improve re-ranking scores, we should go and verify whether or not re-ranking scores correlate with quality and be able to iterate confidently knowing that we might be able to improve the final outcome.
+
+## Negative results are still results
+
+We're not in academia. A negative result is still a result. The goal isn't to publish novel research, the goal is to figure out how to prioritize our limited resources. Remember that to make a decision is to cut off. If we get a negative result or a neutral result, then the outcome is the same, we have made a decision. We have made a decision to cut off this line of inquiry, maybe not forever, but at least for now.
+
+It's also important to trust your judgment too. Just because we're going to cut off this line of reasoning now, it's still good to write up a little memo, explain what happened, write down other things we may not have considered to answer the question, "Under what conditions will we revisit this line of inquiry?"
+
+## Art vs. science
+
+Many people transitioning from classical software engineering to machine learning are often surprised by the empirical nature of the results we obtain. Instead of always having unit tests, we are sampling from the distribution of potential inputs and building a internal model how this system operatess.
```

Comment:
```suggestion
Many people transitioning from classical software engineering to machine learning are often surprised by the empirical nature of the results we obtain. Instead of executing discrete unit tests, we sample from the distribution of potential inputs and build a internal model of how this system operates.
```